sdk/data_obfuscator.go

```diff
@@ -0,0 +1,135 @@
+package sdk
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/ohler55/ojg/jp"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.11.0"
+	"golang.org/x/exp/slices"
+)
+const lengthToObfuscatedByteArray = 8
+var DATA_TO_OBFUSCATE = []string{"http.request.body", "http.response.body", "db.query_result", string(semconv.DBStatementKey), "messaging.payload", "faas.event", "faas.res"}
+var hMacKey []byte
+
+func parseMap(aMap map[string]interface{}) (any,bool) {
+	var result any
+	var changed bool
+	for _, val := range aMap {
+		switch val.(type) {
+		case map[string]interface{}:
+			result, changed = parseMap(val.(map[string]interface{}))
+		case []interface{}:
+			result, changed = parseArray(val.([]interface{}))
+		default:
+			result, changed = obfuscatePrimitives(val)
+		}
+	}
+	return result, changed
+}
+
+func parseArray(anArray []interface{}) (any, bool) {
+	var result any
+	var changed bool
+	for _, val := range anArray {
+		switch val.(type) {
+		case map[string]interface{}:
+			result, changed = parseMap(val.(map[string]interface{}))
+		case []interface{}:
+			result, changed = parseArray(val.([]interface{}))
+		default:
+			result, changed = obfuscatePrimitives(val)
+		}
+	}
+	return result, changed
+}
+
+func modifyElement(element any) (any, bool) {
+	var result any
+	var changed bool
+	switch element.(type) {
+	case map[string]interface{}:
+		result, changed = parseMap(element.(map[string]interface{}))
+	case []interface{}:
+		result, changed = parseArray(element.([]interface{}))
+	default:
+		result, changed = obfuscatePrimitives(element)
+	}
+
+	return result, changed
+}
+
+func obfuscatePrimitives(val any) (any, bool) {
+	var bs []byte
+	if val != nil {
```

Comment:
Why not stringify that isn't a string and handle it like a string?
`fmt.Sprintf("%v", val)`